compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -112,6 +112,9 @@ object Completion {
     val completer = new Completer(mode, prefix, pos)
 
     val completions = path match {
+        case Select(qual @ This(_), _) :: _                    =>
+          val scope = if (qual.span.isSynthetic) completer.scopeCompletions else Map.empty
```

Comment:
I would suggest moving this logic into the body of `selectionCompletions`, leaving this match block here as it was - just as simple dispatching

Reply:
I simplified it a bit. Would it be ok to leave it here in this version?

Reply:
ok